### PR TITLE
[INLONG-10194][Sort] Sqlserver connector support audit ID

### PR DESCRIPTION
**Pull request by XiaoYou201**
### Prepare a Pull Request

- [INLONG-10194][Sort] Sqlserver connectors support audit ID

- Fixes https://github.com/apache/inlong/issues/10194

- parent issue: https://github.com/apache/inlong/issues/8803

### Motivation

Sqlserver connector not support audit function. So this pr will make it support audit function.

### Modifications

Modify Sqlserver connector source.when source/sink receive data, it will send audit information at same time.Some implement are as follows:

Program will get the audit information from the incoming sql statement and pass it to the deserialization classes. During  deserialization, the program will report the number of data received.
企业微信截图_e874eff6-387b-4dda-abd4-201fc3c5b3a2

### CODE REVIEW

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/SqlServerTableSource.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+        DebeziumDeserializationSchema<RowData> deserializer =
+                RowDataDebeziumDeserializeSchema.newBuilder()
+                        .setPhysicalRowType(physicalDataType)
+                        .setMetadataConverters(metadataConverters)
+                        .setResultTypeInfo(typeInfo)
+                        .setServerTimeZone(serverTimeZone)
+                        .setUserDefinedConverterFactory(
+                                SqlServerDeserializationConverterFactory.instance())
+                        .setSourceMetricData(metricOption)
```
Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/SqlServerTableSource.java:137]: method name doesn't match it's variable

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/SqlServerTableSource.java]: fix it

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java [hunk 2]
```diff
@@ -0,0 +1,672 @@
...
+
+/**
+ * Deserialization schema from Debezium object to Flink Table/SQL internal data structure {@link
+ * RowData}.
+ * <p>
+ * Copy from com.ververica:flink-connector-debezium:2.3.0
+ */
+public final class RowDataDebeziumDeserializeSchema
+        implements
+            DebeziumDeserializationSchema<RowData> {
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java:73]:
```suggestion
public final class RowDataDebeziumDeserializeSchema implements DebeziumDeserializationSchema<RowData> {
```

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java [hunk 3]
```diff
@@ -0,0 +1,672 @@
...
+    @Override
+    public void deserialize(SourceRecord record, Collector<RowData> out) throws Exception {
+        Envelope.Operation op = Envelope.operationFor(record);
+        Struct value = (Struct) record.value();
+        Schema valueSchema = record.valueSchema();
+        if (op == Envelope.Operation.CREATE || op == Envelope.Operation.READ) {
+            GenericRowData insert = extractAfterRow(value, valueSchema);
+            validator.validate(insert, RowKind.INSERT);
+            insert.setRowKind(RowKind.INSERT);
+            out = new MetricsCollector<>(out, sourceMetricData);
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java:143]: sourceMetricData may be null, if so, it's not necessary to create a new `MetricsCollector`

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java [hunk 4]
```diff
@@ -0,0 +1,672 @@
...
+                GenericRowData before = extractBeforeRow(value, valueSchema);
+                validator.validate(before, RowKind.UPDATE_BEFORE);
+                before.setRowKind(RowKind.UPDATE_BEFORE);
+                emit(record, before, out);
+            }
+
+            GenericRowData after = extractAfterRow(value, valueSchema);
+            validator.validate(after, RowKind.UPDATE_AFTER);
+            after.setRowKind(RowKind.UPDATE_AFTER);
+            out = new MetricsCollector<>(out, sourceMetricData);
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java:161]: ditto

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java]: fix it, thx~

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java (hunk 3 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java]: fix it, thx~

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java (hunk 2 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java]: fix it, thx~

File: licenses/inlong-sort-connectors/LICENSE [hunk 5]
```diff
@@ -850,6 +850,11 @@
     Source  : org.apache.flink:flink-connector-pulsar:4.1.0-1.18 (Please note that the software have been modified.)
     License : https://github.com/apache/flink-connector-pulsar/blob/main/LICENSE
 
+1.3.24 inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/sqlserver-cdc/src/main/java/org/apache/inlong/sort/sqlserver/RowDataDebeziumDeserializeSchema.java
```
Comment (aloyszhang) [inline licenses/inlong-sort-connectors/LICENSE:853]: Keep consistency of indentation.

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**